core/src/main/java/org/radargun/config/Cluster.java

```diff
@@ -53,23 +53,27 @@ public int getSize() {
    }
 
    public Group getGroup(int slaveIndex) {
-      int index = slaveIndex;
-      for (Group g : groups) {
-         if (index < g.size) {
-            return g;
+      if (slaveIndex != -1) {
```

Comment:
OK, if there are other places in the code that use a default value besides `-1`, I am happy to change it. Can you point me to the other places where there are 'unset' values?